components/ws-manager/pkg/manager/create.go

```diff
@@ -400,6 +422,32 @@ func (m *Manager) createDefiniteWorkspacePod(startContext *startWorkspaceContext
 		}
 	}
 
+	if req.Type == api.WorkspaceType_IMAGEBUILD {
+		pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
+			Name: "gitpod-ca-certificate",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: "ws-manager-tls",
+					Items: []corev1.KeyToPath{
+						{Key: "ca.crt", Path: "ca.crt"},
+					},
+				},
+			},
+		}, corev1.Volume{
+			Name: "image-builder-mk3-config",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
```

Comment:
Until this very PR workspaces have never mounted a configMap or a secret. That's mainly due to performance and reliability, where we've seen "configMap reconciliation loop" failures in the ~past~ recently.

If we were to introduce this, we must also actively observe the behaviour in prod. My gut feeling is that we'd see more image build failures.

Reply:
> Until this very PR workspaces have never mounted a configMap or a secret. That's mainly due to performance and reliability, where we've seen "configMap reconciliation loop" failures in the past recently.

Maybe use `supervisor` to inject the files?

Reply:
Indeed that would be a path. We could just inject that as env var.  Feels a bit odd to build a second mechanism next to configMaps though.

I wonder if there's a good way to get some data on this, or make a more informed decision as to what the impact would actually be. One path could be to just roll with this, try it out and see if it actually fails rather than get apprehensive due to vague past experience :)